src/io/metadata.cpp

```diff
@@ -253,11 +258,41 @@ void Metadata::LoadInitialScore() {
 
   Log::Info("Start loading initial scores");
   num_init_score_ = static_cast<data_size_t>(reader.Lines().size());
-  init_score_ = new float[num_init_score_];
+
+  if (num_init_score_ <= 0){
+      return ;
+  }
+  
+  std::vector<std::string> oneline_init_score = Common::Split(reader.Lines()[0].c_str(), '\t');
+  if (num_class_ != static_cast<int>(oneline_init_score.size())) {
+    Log::Fatal("Number of classes in configuration and initial score file do not match.");    
+  }
+  
+  init_score_ = new score_t[num_init_score_ * num_class_];
   double tmp = 0.0f;
-  for (data_size_t i = 0; i < num_init_score_; ++i) {
-    Common::Atof(reader.Lines()[i].c_str(), &tmp);
-    init_score_[i] = static_cast<float>(tmp);
+  
+  if (num_class_ == 1){
+      for (data_size_t i = 0; i < num_init_score_; ++i) {
+        Common::Atof(reader.Lines()[i].c_str(), &tmp);
+        init_score_[i] = static_cast<score_t>(tmp);
+      }
+  } else {
+      // first line
+      for (int k = 0; k < num_class_; ++k) {
+          Common::Atof(oneline_init_score[k].c_str(), &tmp);
+          init_score_[k * num_data_] = static_cast<score_t>(tmp);
+      }
+      // remaining lines
+      for (data_size_t i = 1; i < num_init_score_; ++i) {
+        oneline_init_score = Common::Split(reader.Lines()[i].c_str(), '\t');  
+        if (static_cast<int>(oneline_init_score.size()) != num_class_){
+            Log::Fatal("Invalid initial score file. Redundant or insufficient columns.");
+        }
+        for (int k = 0; k < num_class_; ++k) {
+          Common::Atof(oneline_init_score[k].c_str(), &tmp);
+          init_score_[k * num_data_ + i] = static_cast<score_t>(tmp);
+        }
+      }
```

Comment:
1. score_t -> float
2. why need to process first line independently? because you have read this before? I think this cost is small. You can just process them in the loop start with '0'